### DISCUSSION TITLE
Manual / Documentation additions

### DISCUSSION THREAD

**Discussion by AlienDiva** [corrados/edrumulus#95] [General]
Hi y'all, starting this thread for information that might be useful or could be included in the project's documentation.

I needed a place document (or dump rantings of) any substantial findings during my build process.

**Comment by AlienDiva** [comment]
_cannot use real-time scheduling (rr/5) (1: operation not permitted) jackd_

This error was produced during call to jackd from run_edrumulus.sh

NOTE: limits.d setting of real time scheduling (rtprio) is apparently no longer effective with recent kernel versions (6.x.x) which includes latest armbian releases.

By disabling cgroup scheduling control fixes permission errors.

`$ sudo sysctl -w kernel.sched_rt_runtime_us=-1`

To make the change permanent.

`$ sudo tee /etc/sysctl.d/disable-cgroup.conf <<< 'kernel.sched_rt_runtime_us=-1'
`

**Reply by AlienDiva** [reply]
For older distributions, the process for enabling real time scheduling use PAM / limits.d as follows:

```
cd /etc/security/limits.d/
sudo mv audio.conf.disabled audio.conf
```
If you don't have a /etc/security/limits.d/audio.conf, create a file named /etc/security/limits.conf and add the following:

```
@audio   -  rtprio     95
@audio   -  memlock    unlimited
```

Then add yourself to the audio group:

`sudo usermod -aG audio $USER`


More info here:
https://jackaudio.org/faq/linux_rt_config.html

**Comment by corrados** [comment]
That is a good idea to document your progress. Regarding the real-time schedulung, you may have already seen it, I have a short notice in the script about this:
https://github.com/corrados/edrumulus/blob/4f7e6160a5fb9de9a59bb97bc656746298214c0c/tools/run_edrumulus.sh#L7
But a better place for all these informations would be this document: https://github.com/corrados/edrumulus/blob/main/doc/manual.md
Right now this document is still work-in-progress.

**Reply by AlienDiva** [reply]
Yes I found your comment in run_edrumulus.sh after already having a solution. I will read through the code more and drop any nuggets of info here as to make it easier to find info with intention to produce a detailed how-to for SBC module builds. Documentation is thin, that ok it's something I can help with as to show appreciation of your efforts.
